test/libsolidity/syntaxTests/abiEncoder/abi_encodeCall_error_free_function.sol

```diff
@@ -0,0 +1,7 @@
+error E(uint);
+
+function f() {
+    abi.encodeCall(E, (1));
+}
+// ----
+// TypeError 3509: (50-51): Expected regular external function type, or external view on public function. Cannot use special function.
```

Comment:
I think we should have a separate message for errors and events. Not sure if technically we can classify them as "special functions" or not, but I think saying event/error would be much clearer for users anyway. Looks like it should be easy to change in `TypeChecker.cpp`, just above the change you already made.

Reply:
I added separate messages for events and errors. Let me know what you think.